x-pack/plugins/ml/readme.md

```diff
@@ -0,0 +1,137 @@
+# Documentation for ML UI developers
+
+## Setup local environment
+
+### Kibana
+
+1. Fork and clone the [Kibana repo](https://github.com/elastic/kibana).
+
+1. Install `nvm`, `node`, `yarn` (for example, by using Homebrew). See 
+   [Install dependencies](https://www.elastic.co/guide/en/kibana/master/development-getting-started.html#_install_dependencies).
+
+1. Make sure that Elasticsearch is deployed and running on localhost:9200.
+
+1. Navigate to the directory of the `kibana` repository on your machine.
+
+1. Fetch the latest changes from the repository.
+
+1. Checkout the branch of the version you want to use. For example, if you want 
+   to use a 7.9 version, run `git checkout 7.9`.
+
+1. Run `nvm use`. The response shows the Node version that the environment uses. 
+   If you need to update your Node version, the response message contains the 
+   command you need to run to do it.
+
+1. Run `yarn kbn bootstrap`. It takes all the dependencies in the code and 
+   installs/checks them. It is recommended to use it every time when you switch 
+   between branches.
+
+1. Make a copy of `kibana.yml` and save as `kibana.dev.yml`. (Git will not track
+   the changes in `kibana.dev.yml` but yarn will use it.)
+   
+1. Provide the appropriate password and user name in `kibana.dev.yml`.
+
+1. Run `yarn start` to start Kibana.
+
+1. Go to http://localhost:560x/xxx (check the terminal message for the exact 
+   path).
+
+
+### Adding sample data to Kibana
+
+Kibana has sample data sets that you can add to your setup so that you can test 
+different configurations on sample data.
+
+1. Click the Elastic logo in the upper left hand corner of your browser to 
+   navigate to the Kibana home page.
+
+1. Click *Load a data set and a Kibana dashboard*.
+
+1. Pick a data set or feel free to click *Add* on all of the available sample 
+   data sets.
+
+These data sets are now ready be analyzed in ML jobs in Kibana.
+
+
+## Running tests
+
+### Jest tests
+
+Run the test following jest tests from `kibana/x-pack`.
+
+New snapshots, all plugins:   
+
+```
+node scripts/jest
+```
+ 
+Update snapshots for the ML plugin: 
+
+```
+node scripts/jest plugins/ml -u
+```
+
+Update snapshots for a specific directory only: 
+
+```
+node scripts/jest plugins/ml/public/settings/filter_lists
```

Comment:
This path is out of date - should be `plugins/ml/public/application/settings/filter_lists` (with the adding application folder level).